src/axom/quest/detail/MarchingCubesImpl.hpp

```diff
@@ -0,0 +1,763 @@
+                                        axom::mint::CELL_CENTERED,
+                                        DIM);
+    }
+
+    const axom::IndexType addedCellCount = m_contourCellCorners.size();
+    const axom::IndexType addedNodeCount = m_contourNodeCoords.size();
+    if(addedCellCount != 0)
+    {
+      const axom::IndexType priorCellCount = mesh.getNumberOfCells();
+      const axom::IndexType priorNodeCount = mesh.getNumberOfNodes();
+      mesh.reserveNodes(priorNodeCount + addedNodeCount);
+      mesh.reserveCells(priorCellCount + addedCellCount);
+
+      mesh.appendNodes((double*)m_contourNodeCoords.data(),
+                       m_contourNodeCoords.size());
+      for(int n = 0; n < addedCellCount; ++n)
+      {
+        MIdx cornerIds = m_contourCellCorners[n];
+        add_to_StackArray(cornerIds, priorNodeCount);
+        mesh.appendCell(cornerIds);
+      }
+      axom::IndexType numComponents = -1;
+      axom::IndexType* dstPtr =
+        mesh.getFieldPtr<axom::IndexType>(cellIdField,
+                                          axom::mint::CELL_CENTERED,
+                                          numComponents);
+      SLIC_ASSERT(numComponents == DIM);
+      axom::ArrayView<axom::StackArray<axom::IndexType, DIM>> dstView(
+        (axom::StackArray<axom::IndexType, DIM>*)dstPtr,
+        priorCellCount + addedCellCount);
+      for(axom::IndexType i = 0; i < addedCellCount; ++i)
+      {
+        dstView[priorCellCount + i] =
+          multidim_cell_index(m_contourCellParents[i]);
+      }
+    }
+  }
+
+  //!@brief Interpolate for the contour location crossing a parent edge.
+  template <int TDIM = DIM>
+  AXOM_HOST_DEVICE typename std::enable_if<TDIM == 2>::type linear_interp(
+    int edgeIdx,
+    const Point cornerCoords[4],
+    const double nodeValues[4],
+    Point& crossingPt)
+  {
+    // STEP 0: get the edge node indices
+    // 2 nodes define the edge.  n1 and n2 are the indices of
+    // the nodes w.r.t. the square or cubic zone.  There is a
+    // agreed-on ordering of these indices in the arrays xx, yy,
+    // zz, nodeValues, crossingPt.
+    int n1 = edgeIdx;
+    int n2 = (edgeIdx == 3) ? 0 : edgeIdx + 1;
+
+    // STEP 1: get the fields and coordinates from the two points
+    const double f1 = nodeValues[n1];
+    const double f2 = nodeValues[n2];
+
+    const Point& p1 = cornerCoords[n1];
+    const Point& p2 = cornerCoords[n2];
+
+    // STEP 2: check whether the interpolated point is at one of the two corners.
+    if(axom::utilities::isNearlyEqual(m_contourVal, f1) ||
+       axom::utilities::isNearlyEqual(f1, f2))
+    {
+      crossingPt = p1;  // memcpy(crossingPt, p1, DIM * sizeof(double));
+      return;
+    }
+
+    if(axom::utilities::isNearlyEqual(m_contourVal, f2))
+    {
+      crossingPt = p2;  // memcpy(crossingPt, p2, DIM * sizeof(double));
+      return;
+    }
+
+    // STEP 3: point is in between the edge points, interpolate its position
+    constexpr double ptiny = 1.0e-80;
```

Comment:
Please use the `PRIMAL_TINY` constant:  
https://github.com/LLNL/axom/blob/develop/src/axom/primal/constants.hpp

Reply:
Done.